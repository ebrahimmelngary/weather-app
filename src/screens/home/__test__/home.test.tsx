import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';

import Home from '../Home';

describe('Home component', () => {
  it('should match snapshot', () => {
    const {toJSON} = render(<Home />);
    expect(toJSON()).toMatchSnapshot();
  });
  it('renders the search input', () => {
    const {getByPlaceholderText} = render(<Home />);
    const input = getByPlaceholderText('Search for a city');
    expect(input).toBeDefined();
  });

  it('updates the search text when input is changed', () => {
    const {getByPlaceholderText, getByTestId} = render(<Home />);
    const input = getByPlaceholderText('Search for a city');
    fireEvent.changeText(input, 'Cairo');
    expect(getByTestId('inputId').props.value).toBe('Cairo');
  });

  it('displays the weather status', () => {
    const {getByTestId, getByText} = render(<Home />);
    expect(getByTestId('weather')).toBeDefined();
    expect(getByTestId('weatherImage')).toBeDefined();
  });
});
