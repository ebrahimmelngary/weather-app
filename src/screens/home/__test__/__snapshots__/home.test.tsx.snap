// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should match snapshot 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    {
      "flexGrow": 1,
      "padding": 16,
    }
  }
>
  <Image
    source={
      {
        "testUri": "../../../src/assets/images/bg.jpg",
      }
    }
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      {
        "paddingVertical": 50,
      }
    }
  >
    <Text
      style={
        {
          "color": "#ffffff",
          "fontSize": 18,
          "fontWeight": "bold",
          "marginBottom": 8,
        }
      }
      testID="weather"
    >
      Weather
    </Text>
    <TextInput
      onChangeText={[Function]}
      placeholder="Search for a city"
      style={
        {
          "backgroundColor": "#ffffff",
          "borderRadius": 5,
          "padding": 10,
        }
      }
      testID="inputId"
      value=""
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "justifyContent": "center",
        "marginVertical": 25,
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
          "padding": 16,
        }
      }
    >
      <Text
        style={
          {
            "color": "#ffffff",
            "fontSize": 18,
            "fontWeight": "500",
            "marginHorizontal": 8,
            "textAlign": "center",
          }
        }
      />
      <Text
        style={
          {
            "color": "#ffffff",
            "fontSize": 18,
            "fontWeight": "500",
            "marginHorizontal": 8,
            "textAlign": "center",
          }
        }
      />
    </View>
    <Image
      source={
        {
          "uri": "https:undefined",
        }
      }
      style={
        {
          "height": 100,
          "width": 100,
        }
      }
      testID="weatherImage"
    />
    <Text
      style={
        {
          "color": "#ffffff",
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    />
  </View>
  <RCTScrollView
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    testID="ForecastList"
    viewabilityConfigCallbackPairs={[]}
    windowSize={5}
  >
    <View />
  </RCTScrollView>
</View>
`;
